packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -613,6 +618,26 @@ export default class CodeGen {
                     }
                 }
 
+                const attributeExpressions = [];
+                for (const attribute of node.attributes) {
+                    const { name, value } = attribute;
+                    if (isExpression(value)) {
+                        this.staticExpressionMap.set(attribute, `a${partId}:${name}`);
```

Comment:
Plan to stash all the tokens at this point and they will all follow a similar format:

`{type}{partId}:{rest}`

a -> attribute
c -> class
s -> style
t -> text

Only attributes will have a value following the `:` since we need to identify which attribute to map it to.